CHANGELOG/CHANGELOG-2.3.md

```diff
@@ -0,0 +1,381 @@
+# Release notes for v2.3.0
+
+[Documentation](https://kubernetes-csi.github.io)
+
+# Changelog since v2.2.2
+
+## Changes by Kind
+
+### Feature
+
+- Add command line arguments to configure leader election options ([#643](https://github.com/kubernetes-csi/external-provisioner/pull/643), [@RaunakShah](https://github.com/RaunakShah))
+- Adds mappings for PV access modes to new CSI access modes: `SINGLE_NODE_SINGLE_WRITER` and `SINGLE_NODE_MULTI_WRITER`. ([#630](https://github.com/kubernetes-csi/external-provisioner/pull/630), [@chrishenzie](https://github.com/chrishenzie))
+- The provisioner sidecar now has an argument called `controller-publish-readonly` which sets the value of CSI PV spec `readonly` field value based on the PVC access mode. If this flag is set to `true` and the PVC access mode only contains the `ROX` access mode, the controller automatically sets `PersistentVolume.spec.CSIPersistentVolumeSource.readOnly` field to `true`. ([#469](https://github.com/kubernetes-csi/external-provisioner/pull/469), [@humblec](https://github.com/humblec))
+- Update external-snapshotter dependency to v4, enabling use of the volume snapshot v1 client ([#650](https://github.com/kubernetes-csi/external-provisioner/pull/650), [@xinyanw409](https://github.com/xinyanw409))
+- Updates Kubernetes dependencies to v1.22.0 ([#660](https://github.com/kubernetes-csi/external-provisioner/pull/660), [@chrishenzie](https://github.com/chrishenzie)) [SIG Storage]
+- Updates container-storage-interface dependency to v1.5.0 ([#644](https://github.com/kubernetes-csi/external-provisioner/pull/644), [@chrishenzie](https://github.com/chrishenzie))
+
+### Bug or Regression
+
+- Fix a data race in cloning protection controller ([#651](https://github.com/kubernetes-csi/external-provisioner/pull/651), [@tksm](https://github.com/tksm))
+
+## Dependencies
+
+### Added
```

Comment:
@chrishenzie I am wondering do we need to list the entire list of dependencies in the changelog? may be just mentioning the main dependency update for example, kube and clients , CSI spec is good enough?

Reply:
ok. please ignore this comment.. Thanks